.github/workflows/check-outdated-content.yml

```diff
@@ -0,0 +1,159 @@
+    - name: Detecting pending translations
+      shell: bash
+      run: |
+        ##### DEBUG section, this will be removed later ###########
+        ls -al
+        git status
+        git branch
+        
+        # Default environment variables 
+        echo "GITHUB_REF: $GITHUB_REF"
+        echo "Extract branch: ${GITHUB_REF#refs/}"
+        
+        # `github` context information
+        echo "(DEBUG) github.ref: ${{github.ref}}"
+        echo "(DEBUG) github.head_ref: ${{github.head_ref}}"
+        echo "(DEBUG) github.base_ref: ${{github.base_ref}}"
+        #####################################################
+
+        OUTPUT_DIR="./outdated"
+        CONTENT_DIR="website/content"
+        languages=("es" "zh")
+        base_lang="en"
+        cd $CONTENT_DIR
+
+        echo "(DEBUG) Current folder: "$(pwd)
+
+        for lang in ${languages[@]}; do
+            touch $lang.txt
+        done
+
+        touch $base_lang.txt
+
+        # Make an output directory
+        if [[ ! -e $OUTPUT_DIR ]]; then
+            mkdir $OUTPUT_DIR
+        elif [[ ! -d $OUTPUT_DIR ]]; then
+            echo "$OUTPUT_DIR already exists but is not a directory" 1>&2
+        fi
+
+        LATEST_BRANCH=${GITHUB_REF#refs/}
+        echo "(DEBUG) LATEST_BRANCH: ${LATEST_BRANCH}"
+
+        # Get the old branch from 'github.base_ref' 
+        # The old branch can be 'upstream/dev-ko'
+        OLD_BRANCH="origin/${{github.base_ref}}"
+        echo "(DEBUG) OLD_BRANCH: ${OLD_BRANCH}"
+            
+        compare () {
+            FILE_PATH=$1
+            # Actually compare between the old and latest English content and log diff in the file
+            if [[ -f "${FILE_PATH}" ]]; then
+                # File exists
+                # Check changes
+                git diff ${OLD_BRANCH}..${LATEST_BRANCH} -- ${FILE_PATH} > temp.diff
+                if [[ -s "temp.diff" ]]; then
+                    echo "(DEBUG) ${FILE_PATH} is outdated."
+                    mkdir -p ${OUTPUT_DIR}/${FILE_PATH%/*}
+                    mv temp.diff ${OUTPUT_DIR}/${FILE_PATH}
+                else 
+                    echo "check if ${FILE_PATH} exist in other languages"
+                    for lang in ${languages[@]}; do
+                        NEW_FILE_PATH=$(echo "${FILE_PATH}" | sed -e "s/${base_lang}\//${lang}\//g")
+                        if [[ ! -e "${NEW_FILE_PATH}" ]]; then
+                            echo "The file ${FILE_PATH} needs to be translated to $lang"
+                            echo ${NEW_FILE_PATH} >> $lang.txt
+                        fi
+                    done
+                fi
+            else
+                echo "(DEBUG) ${FILE_PATH} does not exist."
+            fi
+        }
+
+        find $base_lang -iname "*.md" > files.txt
+        sort files.txt > files_temp.txt;mv files_temp.txt files.txt
+        INFILE=files.txt
+
+        while IFS= read -r line
+        do
+          compare "$line"
+        done < "$INFILE"
```

Comment:
This part makes me think if we may want to move the logic to smth like Go or Python, which handles more complex logic better IMO. Thoughts?

Reply:
I think that it could be less lines with bash, but let me document the script.

Reply:
To get things going I think that it's a non-critical issue and can be an improvement that can be added in a subsequent PR. The work will be tracked in #429 , therefore resolving this thread.